loopy/target/pyopencl.py

```diff
@@ -768,12 +780,67 @@ def get_kernel_call(self,
         if not lsize:
             lsize = (1,)
 
-        value_arg_code, arg_idx_to_cl_arg_idx, cl_arg_count = \
-            generate_value_arg_setup(codegen_state.kernel, skai.passed_names)
+        from loopy.target.pyopencl import split_args_for_overflow
+
+        assert isinstance(kernel.target, PyOpenCLTarget)
+        regular_arg_names, struct_overflow_arg_names = split_args_for_overflow(
+                kernel, skai.passed_names,
+                limit_arg_size_nbytes=kernel.target.limit_arg_size_nbytes,
+                pointer_size_nbytes=kernel.target.pointer_size_nbytes)
+
+        value_arg_code = generate_value_arg_setup(
+                codegen_state.kernel, regular_arg_names)
         arry_arg_code = generate_array_arg_setup(
-            codegen_state.kernel, skai.passed_names, arg_idx_to_cl_arg_idx)
+                codegen_state.kernel, regular_arg_names)
+
+        if struct_overflow_arg_names:
+            regular_arg_names_set = frozenset(regular_arg_names)
+            struct_overflow_arg_names_set = frozenset(
+                    struct_overflow_arg_names)
+
+            py_passed_args = []
+            struct_pack_types = []
+            struct_pack_args = []
+
+            for arg_name in skai.passed_names:
+                if arg_name in regular_arg_names_set:
+                    py_passed_args.append(arg_name)
+                else:
+                    assert arg_name in struct_overflow_arg_names_set
+
+                    arg = kernel.get_var_descriptor(arg_name)
+                    if isinstance(arg, ValueArg):
+                        struct_pack_types.append(arg.dtype.numpy_dtype.char)
+                        struct_pack_args.append(arg_name)
+                    elif isinstance(arg, (ArrayArg, ConstantArg, TemporaryVariable)):
+                        struct_pack_types.append("P")
+                        struct_pack_args.append(f"{arg_name}.svm_ptr")
```

Comment:
Should this be `int_ptr`?

```suggestion
                        struct_pack_args.append(f"{arg_name}.int_ptr")
```

Reply:
No, I changed it to `svm_ptr`. See [here](https://github.com/inducer/pyopencl/blob/5621f2a70a965e570b564ee4e56670912a02d4c3/src/wrap_cl_part_2.cpp#L301-L306).

Reply:
Ah I see, thanks!